presto-tests/src/test/java/com/facebook/presto/Retry.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto;
+
+import org.testng.IRetryAnalyzer;
+import org.testng.ITestResult;
+
+public class Retry
+        implements IRetryAnalyzer
+{
+    private static final int maxTry = 5;
+    private int count;
+
+    @Override
+    public boolean retry(ITestResult iTestResult)
```

Comment:
I am fine adding a RetryAnalyzer for these types of tests.

However, can we add an Exception that indicates that this was flakiness due to the integration test environment?  Basically, can we differentiate flat out wrong results, vs. unexpected state that could potentially be due to the environment?

Reply:
For this particular test, I think it would be really difficult to differentiate directly. This test was added to validate whether the queued query JMX metric is getting incremented/decremented as expected. Before the PR which added the fix for Queued Query JMX metric, the metric was almost always 0. 

Now if the assertion in this test fails, it would be difficult to differentiate in one time failure whether there is actually an issue present or it was just because of a race condition as the test queries are continuously changing states.

Maybe we can also add an assertion for validating that the metric is not zero if the tracker has queries in queued state.

I will give it some more thought to see if I get any other ideas.